test/util/blobfactory/test_util.go

```diff
@@ -74,28 +75,25 @@ func CreateRawTx(txConfig client.TxConfig, msg sdk.Msg, signer *blobtypes.Keyrin
 }
 
 // GenerateRandomAmount generates a random amount for a Send transaction.
-func GenerateRandomAmount() int64 {
-	n := tmrand.Int64()
+func GenerateRandomAmount(rand *tmrand.Rand) int64 {
+	n := rand.Int64()
 	if n < 0 {
 		return -n
 	}
 	return n
 }
```

Comment:
nit: I feel like we could probably just use `int64(rand.Uint64())` instead of this func

Reply:
Thanks for your comment. However, since the max value of `uint64` is almost twice as large as the max value of `int64`, half of the number generated by `rand.Uint64()` will not fit in `int64` hence resulting in overflow and negative values. In the end, we will need to check whether the value is negative and change its sign. Here is a simple test for this:

```go
func TestOverflow(t *testing.T) {
	var v int64
	var r = tmrand.Uint64()
	for v = int64(r); v >= 0; v = int64(r) {
		r = tmrand.Uint64()
	}
	fmt.Println(r)
	fmt.Println(v)
	assert.True(t, v < 0)
}
```
The output I got
```
11946841716761810396
-6499902356947741220
```

Reply:
True, okay probs just leave it as is